packages/drivers/routerlicious-driver/src/documentService.ts

```diff
@@ -334,31 +306,6 @@ export class DocumentService
 		// TODO why are we non null asserting here?
```

Comment:
Seems less of a TODO, more of a question someone is asking. Do we know the answer? Can we remove the comments

Reply:
These were added in #21487 by @RishhiB as part of an effort led by @jason-ha to use Semver compliant TS build settings

Reply:
I think it was done due to addition of some config lint rules. If you see the exclamation in the next line in the end, someone is asking why we have non null assertion there as it is not clear at first sight. It is also a TODO as the task would be understand and remove it. We have such comment at many places in the code. I think Rishi added all those.

For now, we cannot remove it until he solves that problem.

Reply:
Pretty much all such comments are fairly odd.
The underlying typescript setting will be turned off shortly and replaced by a linter rule. These cases are questionable.
`endpoints` is a generic index type. There is nothing in the type system that checks that `storageUrl` or `deltaStorageUrl` are actually present and defined. So, they can't be directly assigned to the respective `this` properties without something saying "yeah, we are really confident that these properties are set". That is what the `!` does.
I think the TODO should really be for owners to confirm that expectation. Ideally there would be a type change, but if not then the `!` will work.

Reply:
Messages crossed. @jatgarg, there is nothing that Rishhi can resolve.
These should really read that the follow up is on code owners to fix the typing or make the code robust. Currently it reads as unsafe. Ideally work items would have been filed.
There are some places where such comments and `!` were added more or less overzealously. This is not one of those cases. If you see that pattern with an array index, those changes are all expected to be reverted.